Quaver.Shared/Online/OnlineManager.cs

```diff
@@ -737,7 +737,8 @@ private static void OnJoinedMultiplayerGame(object sender, JoinedGameEventArgs e
 
             CurrentGame = MultiplayerGames[e.GameId];
             CurrentGame.Players.Add(Self.OnlineUser);
-            CurrentGame.PlayerIds.Add(Self.OnlineUser.Id);
+            if (!CurrentGame.PlayerIds.Contains(Self.OnlineUser.Id))
+                CurrentGame.PlayerIds.Add(Self.OnlineUser.Id);
             CurrentGame.PlayerMods.Add(new MultiplayerPlayerMods { UserId = Self.OnlineUser.Id, Modifiers = "0"});
```

Comment:
I think you should perform similar checks for `Players` and `PlayerMods` as well.